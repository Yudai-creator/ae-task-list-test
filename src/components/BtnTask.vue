<template>
    <button :class="`${btnStyles}`">
        <img :class="btnImgStyles" :src="btnImg">
        <p class="lg:block hidden">
            {{ btnName }}
        </p>
    </button>
</template>

<script setup>

const props = defineProps({
    btnStyles: String,
    btnName: String,
    btnImg: String,
    btnImgStyles: String
})

</script>

