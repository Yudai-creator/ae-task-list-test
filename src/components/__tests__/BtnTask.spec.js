import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import BtnTask from '../BtnTask.vue'


describe('BtnTask', () => {
    const wrapper = mount(BtnTask)
    it('should render properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
    it('should have img', () => {
      expect(wrapper.find("img"))
    })
    it('should have text', () => {
        expect(wrapper.find("p"))
    })
  })