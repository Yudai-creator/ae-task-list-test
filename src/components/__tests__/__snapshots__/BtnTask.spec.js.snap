// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BtnTask > should render properly 1`] = `
"<button class=\\"undefined\\"><img class=\\"\\">
  <p class=\\"lg:block hidden\\"></p>
</button>"
`;
